beacon-chain/orchestrator/client.go

```diff
@@ -0,0 +1,57 @@
+package orchestrator
+
+import (
+	"context"
+	"fmt"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+const confirmVanBlockHashesMethod = "orc_confirmVanBlockHashes"
+
+type Client interface {
+	Dial(string) (*Client, error)
+	DialContext(context.Context, string) (*Client, error)
+	NewClient(*rpc.Client) *Client
+	Close()
+	ConfirmVanBlockHashes(context.Context, []*BlockHash) ([]*BlockStatus, error)
+}
+
+// RPCClient defines typed wrappers for the Ethereum RPC API.
+type RPCClient struct {
+	client *rpc.Client
+}
+
+// Dial connects a client to the given URL.
+func Dial(rawurl string) (*RPCClient, error) {
+	return DialContext(context.Background(), rawurl)
+}
+
+func DialContext(ctx context.Context, rawurl string) (*RPCClient, error) {
+	c, err := rpc.DialContext(ctx, rawurl)
+	if err != nil {
+		return nil, err
+	}
+	return NewClient(c), nil
+}
+
+// NewClient creates a client that uses the given RPC client.
+func NewClient(c *rpc.Client) *RPCClient {
+	return &RPCClient{c}
+}
+
+func (orcRpc *RPCClient) Close() {
+	orcRpc.client.Close()
+}
+
+func (orcRpc *RPCClient) ConfirmVanBlockHashes(ctx context.Context, blockHashes []*BlockHash) ([]*BlockStatus, error) {
```

Comment:
@mxmar Method definition should be `ConfirmVanBlockHashes(ctx context.Context, request []*vanTypes.ConfirmationReqData) (response []*vanTypes.ConfirmationResData, err error)` and in this method, you just convert common.Hash to [32] byte for hash

Reply:
Ok, so convertion logic will lie in this method as you has suggested.

Reply:
yes

Reply:
done